### DISCUSSION TITLE
How to compose transformations?

### DISCUSSION THREAD

**Discussion by jmcph4** [jonasbb/serde_with#767] [Q&A] [answered]
Consider two possible JSON responses from an API:

```json
{
    "some_key": ""
}
```

```json
{
    "some_key": "12.33"
}
```

Is there a way to apply both `NoneAsEmptyString` and `DisplayFromStr`?

```rust
#[serde_as]
#[derive(Clone, Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Response {
    #[serde_as(as = "NoneAsEmptyString")]
    #[serde_as(as = "DisplayFromStr")]
    pub some_key: Option<f64>,
}
```

**Comment by jonasbb** [comment, marked as answer]
Unfortunately it is impossible to combine multiple transformations. You can create you own type that implements the behavior of both combined. The documentation has some information how to get started and you can look at the implementation of both types.
https://docs.rs/serde_with/3.8.3/serde_with/guide/serde_as/index.html#implementing-serializeas--deserializeas
